src/LayerEditor/ui/data/block_layers_model.py

```diff
@@ -47,64 +47,22 @@ def get_interface_node_sets(self):
     @property
     def decomposition(self):
         # This is going to be problem in future when there will be more editable layers in one block
-        for layer in self.layers_dict.values():
-            if isinstance(layer.top_in, InterfaceNodeSetItem):
-                return layer.top_in.decomposition
+        return self.get_interface_node_sets()[0].decomposition
 
     @property
     def layer_names(self):
-        for layer in self.get_sorted_layers():
+        """Return unordered set of layer names in block."""
```

Comment:
""" Generate a sequence of layer names in the block, arbitrary order."""

Reply:
Done in newest branch